subsys/bluetooth/host/l2cap.c

```diff
@@ -1278,8 +1274,24 @@ static void le_ecred_reconf_req(struct bt_l2cap *l2cap, uint8_t ident,
 			goto response;
 		}
 
-		BT_L2CAP_LE_CHAN(chan)->tx.mtu = mtu;
-		BT_L2CAP_LE_CHAN(chan)->tx.mps = mps;
+		if (BT_L2CAP_LE_CHAN(chan)->tx.mps > mps) {
+			mps_reduced = true;
+		}
```

Comment:
I don't think we can do that, because we need to check if MPS was reduced at least for one channel, and if the total number of channels is at least 2 (`mps_reduced && i > 1`). So we need to do this after while loop

Reply:
This will return error if MPS was decreased, but only one channel was requested

Reply:
You are correct; forgot the "more than 1 channel" part :)